libexec/bootstrap-scripts/main-deffile.sh

```diff
@@ -53,9 +53,23 @@ if [ -n "${BOOTSTRAP:-}" -a -z "${SINGULARITY_BUILDNOBASE:-}" ]; then
     fi
 fi
 
+# take a snapshot of the environment for later comparison
+if [ ${BOOTSTRAP} = "localimage" ]; then
+    export SINGULARITY_STARTING_ENVIRONMENT=$(eval_abort env -i ${SINGULARITY_libexecdir}/singularity/helpers/record-env.sh ${SINGULARITY_ROOTFS})
+    export SINGULARITY_STARTING_ENVSHA1=$(eval_abort sha1sum ${SINGULARITY_ROOTFS}/.singularity.d/env/*.sh | sha1sum)
+fi 
```

Comment:
oh and I think we would want to do `${BOOTSTRAP-}` so if it's not set the default is empty. I don't totally get it but I see @gmk do it and he's bash shell god so I figure it's the right thing to do :) http://tldp.org/LDP/abs/html/parameter-substitution.html